### DISCUSSION TITLE
person cannot register at our Instance, email is already taken

### DISCUSSION THREAD

**Discussion by Von-Birne** [mastodon/mastodon#19312] [Q&A] [answered]
Hi,

recently a person reached out to me stating that they want to have an account at our instance again, but they cannot register using their email address (which seems to be still in the system).

After reproducing the issue myself I queried the database. There is no entry in the users table having said email.

```
mastodon_production=> select * from users where email ~* '<email address>';
(0 rows)
```

I have no clue where to look further. A potential workaround to alter the email by adding a + sadly did not work. Any idea where I can find the email address and how to resolve the issue?

Thank you!

**Comment by Gargron** [comment, marked as answer]
It is possible that the person was previously suspended, in which case a hash of the e-mail has been saved in the system to prevent them from registering again. The thing to check is called a "canonical e-mail block" (because it takes `+` and some other stuff into account)

**Reply by Von-Birne** [reply]
do you know which hash algorithm is used for storing this hash?

Also how likely is it, that this will break stuff and what?

**Reply by Gargron** [reply]
SHA256. It won't break anything if you delete a matching record from `canonical_email_blocks`.

**Reply by Von-Birne** [reply]
I found a hash matching the email. Thank you!
